src/patternfly/components/Tabs/tabs.scss

```diff
@@ -112,167 +126,305 @@
     }
+  }
+
+  // Scroll buttons
+  .pf-c-tabs__scroll-button:nth-of-type(1) {
+    margin-right: 0;
+    border-right-width: var(--pf-c-tabs__button--before--BorderWidth);
+    transform: translateX(0);
+  }
+
+  .pf-c-tabs__scroll-button:nth-of-type(2) {
+    margin-left: 0;
+    border-left-width: var(--pf-c-tabs__button--before--BorderWidth);
+    opacity: 1;
+    transform: translateX(0);
+  }
+
+  &.pf-m-secondary .pf-c-tabs__scroll-button,
+  &.pf-m-no-border .pf-c-tabs__scroll-button {
+    border-bottom-color: transparent;
+    border-bottom-width: 0;
+  }
+
+  .pf-c-tabs__item:first-child .pf-c-tabs__button::before {
+    border-left-width: 0;
+  }
+
+  &.pf-m-no-border .pf-c-tabs__button::before,
+  &.pf-m-no-border .pf-c-tabs__scroll-button::before {
+    border-bottom-width: 0;
+  }
+}
+
+// Box
+.pf-c-tabs.pf-m-box {
+  --pf-c-tabs__button--Background: var(--pf-c-tabs--m-box--Background);
+
+  &::before {
+    top: calc(100% - var(--pf-c-tabs--before--BorderWidth));
+  }
+
+  .pf-c-tabs__button::after {
+    border-top-color: var(--pf-c-tabs__button--after--BorderColor);
+    border-top-width: var(--pf-c-tabs__button--after--BorderWidth);
+    border-bottom-width: 0;
+  }
+
+  .pf-c-tabs__button::before {
+    border-bottom-width: var(--pf-c-tabs__button--before--BorderWidth);
+    border-left-width: var(--pf-c-tabs__button--before--BorderWidth);
+  }
+
+  // stylelint-disable selector-max-class
+  .pf-c-tabs__item.pf-m-current .pf-c-tabs__button {
+    --pf-c-tabs__button--Background: var(--pf-c-tabs__item--m-current--c-tabs__button--Background);
+  }
+
+  .pf-c-tabs__item.pf-m-current .pf-c-tabs__button::before {
+    border-bottom-color: var(--pf-c-tabs__button--Background);
+  }
+
+  .pf-c-tabs__item:not(.pf-m-current):first-child .pf-c-tabs__button::before {
+    border-left-width: 0;
+  }
+
+  .pf-c-tabs__item.pf-m-current:last-child .pf-c-tabs__button::before {
+    border-right-width: var(--pf-c-tabs__button--before--BorderWidth);
+  }
+  // stylelint-enable
+}
+
+// Vertical
+.pf-c-tabs.pf-m-vertical {
+  --pf-c-tabs--Inset: var(--pf-c-tabs--m-vertical--Inset);
+
+  padding: var(--pf-c-tabs--Inset) 0 var(--pf-c-tabs--Inset) 0;
 
-    &:nth-of-type(1) {
-      box-shadow: var(--pf-c-tabs__scroll-button--m-secondary--nth-of-type-1--BoxShadow);
+  &::before {
+    top: 0;
+    bottom: 0;
+    left: 0;
+    border-bottom: none;
+    border-left: var(--pf-c-tabs--before--BorderWidth) solid var(--pf-c-tabs--before--BorderColor);
+  }
+
+  .pf-c-tabs__list {
+    flex-direction: column;
+    border: 0;
+  }
+
+  .pf-c-tabs__button {
+    --pf-c-tabs__button--PaddingTop: var(--pf-c-tabs--m-vertical--c-tabs__button--PaddingTop);
+    --pf-c-tabs__button--PaddingBottom: var(--pf-c-tabs--m-vertical--c-tabs__button--PaddingBottom);
+
+    text-align: left;
+
+    &::after {
+      border-top-width: 0;
+      border-bottom-width: 0;
+      border-left-color: var(--pf-c-tabs__button--after--BorderColor);
```

Comment:
Could all of the places we're overriding these properties be set initially as a CSS var instead, then have vars for the values? It seems like there are a lot of property declarations for both `.pf-c-tabs__button` and `.pf-c-tabs__scroll-button` for the borders that could be vars that are individually themeable.

Reply:
I think that would also enable us to theme the variables in some of these selectors without having to write out so many classnames in the selectors to target individual properties.